kraken-std/src/kraken/std/dist.py

```diff
@@ -264,3 +254,21 @@ def dist(
         archive_type=archive_type,
         prefix=prefix,
     )
+
+
+def get_configured_resources(
+    resources: Dict[Task, list[Resource]],
+    dependencies_map: dict[str, IndividualDistOptions],
+    dependencies_set: TaskSet,
+) -> list[ConfiguredResource]:
+    configured_resources = []
+
+    for task, task_resources in resources.items():
+        for resource in task_resources:
+            task_options = dependencies_map[next(iter(dependencies_set.partitions()[task]))]
+            options = IndividualDistOptions(**vars(task_options))
+
+            if options.arcname is None and isinstance(resource, (BinaryArtifact, LibraryArtifact)):
```

Comment:
the solution is to move the logic where we construct the `ConfiguredResource`s, to set the `arcname` in the `options` there

note that `LibraryArtifact`s now _also_ get their `arcname` changed, which is hopefully not controversial